clientconn.go

```diff
@@ -624,19 +625,30 @@ func (cc *ClientConn) handleResolvedAddrs(addrs []resolver.Address, err error) {
 	cc.mu.Lock()
 	defer cc.mu.Unlock()
 	if cc.conns == nil {
+		// cc was closed.
+		return
+	}
+
+	if reflect.DeepEqual(cc.curAddresses, addrs) {
```

Comment:
I have heard just importing the reflect package is a bad idea (in the form of performance or binary size or both?).  We should look into that claim and remove it if so.  Looks like we're already importing it, so LGTM for this PR.

Reply:
We should try the [`cmp`](https://github.com/google/go-cmp) package.